api_tests/files/views/test_file_metadata_record_detail.py

```diff
@@ -0,0 +1,165 @@
+import pytest
+
+from api.base.settings.defaults import API_BASE
+from api_tests import utils
+from osf_tests.factories import (
+    AuthUserFactory,
+    ProjectFactory,
+    RegistrationFactory,
+)
+
+
+@pytest.mark.django_db
+class TestFileMetadataRecordDetail:
+
+    @pytest.fixture()
+    def user(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def private_record(self, user):
+        private_node = ProjectFactory(creator=user)
+        private_file = utils.create_test_file(private_node, user, filename='private_file')
+        return private_file.records.get(schema___id='datacite')
+
+    @pytest.fixture()
+    def public_record(self, user):
+        public_node = ProjectFactory(creator=user, is_public=True)
+        public_file = utils.create_test_file(public_node, user, filename='public_file')
+        return public_file.records.get(schema___id='datacite')
+
+    def get_url(self, record):
+        return '/{}files/{}/metadata_records/{}/'.format(API_BASE, record.file._id, record._id)
+
+    def test_metadata_record_detail(self, app, user, public_record, private_record):
+
+        # test_unauthenticated_can_view_public_file_metadata_record
+        res = app.get(self.get_url(public_record))
+        assert res.status_code == 200
+        assert res.json['data']['id'] == public_record._id
+
+        # test_authenticated_can_view_public_file_metadata_record
+        res = app.get(self.get_url(public_record), auth=user.auth)
+        assert res.status_code == 200
+        assert res.json['data']['id'] == public_record._id
+
+        # test_unauthenticated_cannot_view_private_file_metadata_record
+        res = app.get(self.get_url(private_record), expect_errors=True)
+        assert res.status_code == 401
+
+        # test_authenticated_can_view_private_file_metadata_record
+        res = app.get(self.get_url(private_record), auth=user.auth)
+        assert res.status_code == 200
+        assert res.json['data']['id'] == private_record._id
+
+        # test_unauthorized_cannot_view_private_file_metadata_record
+        unauth = AuthUserFactory()
+        res = app.get(self.get_url(private_record), auth=unauth.auth, expect_errors=True)
+        assert res.status_code == 403
+
+        # test_cannot_delete_metadata_records
+        res = app.delete_json_api(self.get_url(public_record), auth=user.auth, expect_errors=True)
+        assert res.status_code == 405
+
+
+@pytest.mark.django_db
+class TestFileMetadataRecordUpdate:
+
+    @pytest.fixture()
+    def user(self):
```

Comment:
could DRY this up the teeniest bit by moving the user fixture outside the classes